optax/_src/utils.py

```diff
@@ -67,52 +70,58 @@ def set_diags(a, new_diags):
   return a
 
 
-def multi_normal(loc, log_scale):
-  return MultiNormalDiagFromLogScale(loc=loc, log_scale=log_scale)
-
-
 class MultiNormalDiagFromLogScale():
   """MultiNormalDiag which directly exposes its input parameters."""
 
-  def __init__(self, loc, log_scale):
+  def __init__(self, loc: chex.Array, log_scale: chex.Array):
     self._log_scale = log_scale
     self._scale = jnp.exp(log_scale)
     self._mean = loc
     self._param_shape = jax.lax.broadcast_shapes(
         self._mean.shape, self._scale.shape)
 
-  def sample(self, shape, seed):
+  def sample(self, shape: Sequence[int],
+             seed: jax.random.PRNGKey) -> chex.Array:
     sample_shape = shape + self._param_shape
     return jax.random.normal(
         seed, shape=sample_shape) * self._scale  + self._mean
 
-  def log_prob(self, x):
-    log_prob = multivariate_normal.logpdf(x, loc=self._mean, scale=self._scale)
+  def log_prob(self, x: chex.Array) -> chex.Array:
+    log_prob = multivariate_normal.logpdf(x,
+                                          loc=self._mean,
+                                          scale=self._scale)
     # Sum over parameter axes.
     sum_axis = [-(i + 1) for i in range(len(self._param_shape))]
     return jnp.sum(log_prob, axis=sum_axis)
 
   @property
-  def log_scale(self):
+  def log_scale(self) -> chex.Array:
     return self._log_scale
 
   @property
-  def params(self):
+  def params(self) -> List[chex.Array]:
```

Comment:
I think we should prefer the abstract type Sequence in this case - just so that we don't make any promises where they aren't strictly necessary. We might want to return a tuple or named tuple in the future.